src/olympia/blocklist/mlbf.py

```diff
@@ -148,16 +166,18 @@ def soft_blocked_items(self) -> List[str]:
     def not_blocked_items(self) -> List[str]:
         all_blocks_ids = [version.version_id for version in self._all_blocks]
         not_blocked_items = MLBF.hash_filter_inputs(
-            Version.unfiltered.exclude(id__in=all_blocks_ids or ()).values_list(
-                'addon__addonguid__guid', 'version'
-            )
+            Version.unfiltered.exclude(id__in=all_blocks_ids or ())
+            .order_by('id')
+            .values_list('addon__addonguid__guid', 'version')
         )
         # even though we exclude all the version ids in the query there's an
         # edge case where the version string occurs twice for an addon so we
         # ensure not_blocked_items contain no blocked_items or soft_blocked_items.
-        return list(
-            set(not_blocked_items) - set(self.blocked_items + self.soft_blocked_items)
```

Comment:
I am a bit nervous about getting rid of the sets because of https://github.com/mozilla/addons/issues/15132 but OTOH `not_blocked_items` should only contain unique entries already.

Reply:
I was looking at this and I'm not even sure how I would test it... I added `distinct()` to the queries but that's kind of moot as we have unique indexes for  all the relevant combinations.